lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -302,26 +311,46 @@ void SessionStateEnforcer::m5g_complete_termination(
   return;
 }
 
-/*Removing all associated rules to session*/
-void SessionStateEnforcer::m5g_remove_rules_and_update_upf(
+void SessionStateEnforcer::m5g_move_to_inactive_state(
+    std::string& imsi, std::unique_ptr<SessionState>& session,
+    SetSmNotificationContext notif, SessionStateUpdateCriteria* session_uc) {
+  set_new_fsm_state_and_increment_version(session, INACTIVE, session_uc);
+  /* Call for all rules to be de-associated from session
+   * and inform to UPF
+   */
+  m5g_pdr_rules_change_and_update_upf(imsi, session, PdrState::IDLE);
+  return;
+}
+
+void SessionStateEnforcer::set_new_fsm_state_and_increment_version(
+    std::unique_ptr<SessionState>& session, SessionFsmState targetState,
+    SessionStateUpdateCriteria* session_uc) {
+  auto stateStr = session_fsm_state_to_str(session->get_state());
+  session->set_fsm_state(targetState, session_uc);
+  uint32_t cur_version = session->get_current_version();
+  session->set_current_version(++cur_version, session_uc);
+  MLOG(MDEBUG) << "During " << stateStr << " state of session"
+               << "of imsi: " << session->get_imsi()
+               << "/teid:" << session->get_upf_local_teid() << "changed to "
+               << session_fsm_state_to_str(session->get_state());
+  return;
+}
+
+void SessionStateEnforcer::m5g_pdr_rules_change_and_update_upf(
     const std::string& imsi, const std::unique_ptr<SessionState>& session,
-    const std::string& dnn, SessionStateUpdateCriteria& uc) {
-  // remove all rules;
-  MLOG(MINFO) << "Will be removing all associated rules of session id "
-              << session->get_session_id();
-  auto ip_addr = session->get_config()
-                     .rat_specific_context.m5gsm_session_context()
-                     .pdu_address()
-                     .redirect_server_address();
+    enum PdrState pdrstate) {
+  // update crieria status not needed
+  session->set_all_pdrs(pdrstate);
+  session->reset_rtx_counter();
+  m5g_send_session_request_to_upf(imsi, session);
+  return;
+}
+
+void SessionStateEnforcer::m5g_send_session_request_to_upf(
+    const std::string& imsi, const std::unique_ptr<SessionState>& session) {

```

Comment:
`#include <memory> ` lint below

```
 Add #include <memory> for unique_ptr<>  [build/include_what_you_use] [4]
```

Reply:
Included the header file <memory> as suggested